modules/environment/aws/code-services/main.tf

```diff
@@ -22,21 +22,50 @@ data "aws_subnets" "eks_workers" {
   }
 }
 
-#tfsec:ignore:aws-s3-specify-public-access-block
-#tfsec:ignore:aws-s3-enable-versioning
-#tfsec:ignore:aws-s3-enable-bucket-encryption
-#tfsec:ignore:aws-s3-enable-bucket-logging
 resource "aws_s3_bucket" "code_services_bucket" {
   bucket = "code-services-${var.account_id}-${var.cluster}"
+
+  #Enables encryption using customer managed keys
+  server_side_encryption_configuration {
+    rule {
+      apply_server_side_encryption_by_default {
+        kms_master_key_id = aws_kms_key.code_services_key.arn
+        sse_algorithm     = "aws:kms"
+      }
+    }
+  }
 }
 
+resource "aws_kms_key" "code_services_key" {
+  description             = "This key is used to encrypt bucket objects"
+  deletion_window_in_days = 10
+  enable_key_rotation     = true
+}
+
+
 resource "aws_s3_bucket_versioning" "code_services_versioning" {
   bucket = aws_s3_bucket.code_services_bucket.id
   versioning_configuration {
     status = "Enabled"
   }
 }
 
+resource "aws_s3_bucket_logging" "code_services_logging" {
+  bucket = aws_s3_bucket.code_services_bucket.id
+
+  target_bucket = "code-services-${var.account_id}-${var.cluster}"
```

Comment:
Sending logs to the source bucket is not recommended https://aws.amazon.com/premiumsupport/knowledge-center/s3-server-access-logs-same-bucket/

We should probably setup a dedicated s3 bucket just for logging.